xprof/tally.cpp

```diff
@@ -159,49 +159,50 @@ void btx_finalize_usr_data(void *btx_handle, void *usr_data) {
+				uint64_t min, uint64_t max, uint64_t total, uint64_t count,
+				uint64_t backend, uint64_t err) {
+
   tally_dispatch_t *data = (tally_dispatch_t *)usr_data;
 
-  TallyCoreTime a{dur, (uint64_t)err};
-  const int level = data->backend_level[backend_id];
-  data->host_backend_name[level].insert(backend_name[backend_id]);
-  data->host[level][hpt_function_name_t(hostname, vpid, vtid, name)] += a;
+  const int level = data->backend_level[backend];
+  data->host_backend_name[level].insert(backend_name[backend]);
+  data->host[level][ {hostname, vpid, vtid, name } ] += {total, err, count, min, max};
 }
 
-static void device_usr_callback(void *btx_handle, void *usr_data, const char *hostname,
-                                      int64_t vpid, uint64_t vtid, int64_t ts, int64_t backend,
-                                      const char *name, uint64_t dur, uint64_t did, uint64_t sdid,
-                                      bt_bool err, const char *metadata) {
+static void tally_device_callback(void *btx_handle, void *usr_data, const char *hostname,
+                                int64_t vpid, uint64_t vtid,  const char *name,
+                                uint64_t min, uint64_t max, uint64_t total, uint64_t count,
+				uint64_t did, uint64_t sdid, const char * metadata ) {
+
   tally_dispatch_t *data = (tally_dispatch_t *)usr_data;
 
   const auto name_demangled = (strcmp(data->params->name, "demangle") == 0) ? f_demangle_name(name) : name;
   const auto name_with_metadata = (data->params->display_kernel_verbose && !strcmp(metadata, ""))
                                       ? name_demangled + "[" + metadata + "]"
                                       : name_demangled;
 
-  TallyCoreTime a{dur, (uint64_t)err};
-  data->device[hpt_device_function_name_t(hostname, vpid, vtid, did, sdid, name_with_metadata)] +=
-      a;
+  data->device[ {hostname, vpid, vtid, did, sdid, name_with_metadata} ] += {total, 0, count, min, max};
 }
 
-static void traffic_usr_callback(void *btx_handle, void *usr_data, const char *hostname,
-                                       int64_t vpid, uint64_t vtid, int64_t ts, int64_t backend,
-                                       const char *name, uint64_t size) {
+static void tally_traffic_callback(void *btx_handle, void *usr_data, const char *hostname,
+                                int64_t vpid, uint64_t vtid,  const char *name,
+                                uint64_t min, uint64_t max, uint64_t total, uint64_t count,
+                                uint64_t backend) {
+
   tally_dispatch_t *data = (tally_dispatch_t *)usr_data;
 
-  TallyCoreByte a{(uint64_t)size, false};
   const int level = data->backend_level[backend];
   data->traffic_backend_name[level].insert(backend_name[backend]);
-  data->traffic[level][hpt_function_name_t(hostname, vpid, vtid, name)] += a;
+  data->traffic[level][ {hostname, vpid, vtid, name} ] +=  {total, 0, count, min, max};;
```

Comment:
* Are `total` and `duration` (`dur`) the same? 
* Can we consider the use of `dur` in place of `total` in the code and the model.yaml to be consistent?

Reply:
Hum true, but we cannot really change the model and we will need to change in the `interval.c` too. Let me think

Reply:
Oh, how the change from `total` to `dur` is related to `interval.c`?

Reply:
I mean we have `duration` for the interval and `total` for the interval. Doesn't that make sense?

Reply:
You are right that makes sense.